src/types/mod.rs

```diff
@@ -94,8 +95,8 @@ pub use final_data::{CompleteBound, Final};
 pub enum Error {
     /// An attempt to bind a type conflicted with an existing bound on the type
     Bind {
-        existing_bound: Bound,
-        new_bound: Bound,
+        existing_bound: Type,
+        new_bound: Type,
```

Comment:
7e7756ebcdde18b28ed0a2193380b8d7cc065e5c: I think this change is acceptable. `Type` and `Bound` are conceptually similar.